conda_build/environ.py

```diff
@@ -47,6 +47,10 @@ def get_git_build_info(src_dir, git_url, expected_rev):
     env = os.environ.copy()
     d = {}
     git_dir = join(src_dir, '.git')
+    if not isinstance(git_dir, str):
```

Comment:
OK, on Windows, I see this:

```
>>> import sys
>>> sys.getdefaultencoding()
'ascii'
>>> sys.getfilesystemencoding()
'mbcs'
```

And the [Python docs][1] say:

[1]: https://docs.python.org/2/library/sys.html#sys.getfilesystemencoding

>On Windows NT+, file names are Unicode natively, so no conversion is performed. [**getfilesystemencoding()**][1] still 
>returns `'mbcs'`, as this is the encoding that applications should use when they explicitly want to convert Unicode 
>strings to byte strings that are equivalent when used as file names.

So, I still think using `getfilesystemencoding()` is correct in this case, and using `str(git_dir)` would be wrong, even though it might work in many cases.

But either way, I'm sure that people have probably developed a healthy aversion to putting umlauts in their git repo names.  That's just asking for trouble.